features/steps/misc.py

```diff
@@ -38,3 +53,24 @@ def verify_timer_interval_for_job(context, job, interval):
     run_diff = next_run - last_run
 
     assert_that(run_diff.seconds, equal_to(int(interval)))
+
+
+@when("I prepare the autocomplete test")
+def prepare_autocomplete_test(context):
+    when_i_run_command(context, "apt update", "with sudo")
+    when_i_apt_install(context, "expect")
+    when_i_create_file_with_content(
+        context, "/tmp/test_autocomplete.exp", text=AUTOCOMPLETE_TEST_FILE
+    )
+    when_i_run_command(
+        context, "chmod +x /tmp/test_autocomplete.exp", "with sudo"
+    )
+
+
+@when("I press tab twice to autocomplete the `{command}` command")
+def when_i_autocomplete_command(context, command):
+    when_i_run_command(
+        context,
+        "/tmp/test_autocomplete.exp '{}'".format(command),
+        "as non-root",
+    )
```

Comment:
If we want to include xenial again and maybe make this a little less fragile as we add more subcommands. We could do something like this
```suggestion
    )
    context.autocomplete_suggestions = sorted(context.process.stdout.split())
    
@then("the autocomplete suggestions are")
def then_the_autocomplete_suggestions_are(context):
    assert sorted(context.text.split()) == context.autocomplete_suggestions
```

Then in the feature file, something like:
```gherkin
        And I press tab twice to autocomplete the `ua` command
        Then the autocomplete suggestions are
        """
        --debug
        --help
        --version
        api
        attach
        auto-attach
        collect-logs
        config
        detach
        disable
        enable
        fix
        help
        refresh
        security-status
        status
        system
        version
        """
```

thoughts? Is that too complicated for this? or does it help?

Reply:
1. Not including Xenial is not making us fragile, as we know the output is always the same and ordering is the only difference
2. There is more manipulation that needs to be done in the output, as it would not be just exactly the autocomplete options, but rather the whole shell output

I vote not worth it

Reply:
Sorry I meant "include xenial again" and "make this less fragile" as two separate concerns.

I agree that including xenial isn't that important.

My "fragile" concern was that when we add a new subcommand or flag, then the multi-column suggestion output will be annoying to update - because an addition in the first column will shift all the other values.


> There is more manipulation that needs to be done in the output, as it would not be just exactly the autocomplete options, but rather the whole shell output

I didn't realize that, and I agree that makes this not worth it